catalyst-gateway-crates/c509-certificate/src/c509_cert.rs

```diff
@@ -0,0 +1,282 @@
+//! This module provides essential functions for encoding and generating C509
+//! certificates. Please refer to the [c509-certificate](https://datatracker.ietf.org/doc/draft-ietf-cose-cbor-encoded-cert/09/) for more information.
+
+use minicbor::Encoder;
+use wasm_bindgen::prelude::wasm_bindgen;
+
+use crate::{
+    c509_enum::{C509CertificateType, PubKeyAlgoRegistry, SignatureAlgoRegistry},
+    cbor_encoder::{CborEncoder, Name, Time, UnwrappedBiguint},
+    extensions::{encode_extensions, Extensions},
+};
+
+// ---------------------------------------------------
+
+/// Encodes the subject public key information.
+///
+/// # Arguments
+///
+/// * `pubk_type` - The type of the public key algorithm.
+/// * `pubk` - The public key bytes.
+/// * `encoder` - A mutable reference to a `minicbor::Encoder` to encode the extensions
+///   into.
+///
+/// # Panics
+///
+/// Panics if the public key type is not supported (RSA and Elliptic curve in Weierstraß
+/// public keys).
+///
+/// # References
+///
+/// Please refer to the [c509-certificate](https://datatracker.ietf.org/doc/draft-ietf-cose-cbor-encoded-cert/09/) Section 3.1. Message Fields -> subjectPublicKeyInfo for more information.
+fn encode_subject_public_key(
+    pubk_type: PubKeyAlgoRegistry, pubk: Vec<u8>, encoder: &mut Encoder<&mut Vec<u8>>,
+) {
```

Comment:
Why this function does not return any errors ?
If I am not mistaken `encoder.i8()` and `encoder.bytes()` could return errors.